hack/spellcheck.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+# Spelling errors detected in markdown files.
+# If the errors are names, external links, or unusual but accurate technical words,
+# then you should create an inline comment like:
+#
+# <!-- cSpell:ignore someword someotherword -->
+#
+# Of course, you should only include non-dictionary words that are correctly spelled!
+# If the error happens because of a common technical term or proper name that is likely
+# to appear many times, then please edit "../.cspell-config.json" and add it to the
+# "words" list.
+
+set -eux
+
+IS_CONTAINER="${IS_CONTAINER:-false}"
+CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+
+# all md files, but ignore .github and node_modules
+if [ "${IS_CONTAINER}" != "false" ]; then
+    npm install -g cspell
+    npx cspell lint --show-suggestions -c /workdir/.cspell-config.json ./**/*.md
+else
+    "${CONTAINER_RUNTIME}" run --rm \
+        --env IS_CONTAINER=TRUE \
+        --volume "${PWD}:/workdir:ro,z" \
+        --entrypoint sh \
+        --workdir /workdir \
+        docker.io/library/node:21.6.0-bookworm@sha256:0ded28778059262bd3c066b609186e5b6c89550a9362dce4309ad67c95af0d77 \
```

Comment:
is this the tool we are using https://github.com/streetsidesoftware/cspell-cli/pkgs/container/cspell and if it is why not we use ghcr.io/streetsidesoftware/cspell:8.3.2 instead?

Reply:
The image size is much small + you get a pinned version of cspell
![Screenshot 2024-01-25 at 15 07 51](https://github.com/metal3-io/metal3-io.github.io/assets/52313821/d29d9373-afdc-49d1-adff-d458fc159fef)

Reply:
cspell-cli is different from cspell, but I can check if it achieves the same.

Reply:
yeah but here as you can see I have downloaded the cspell image not the cspell-cli, I am guessing this should have the cspell binary that we are installing in the script currently

Reply:
It might not have everything that we need but worth a check, thanks for checking it.

Reply:
It works the same, so its much better this way. Thanks for the suggestion.